flink-yarn-tests/src/test/java/org/apache/flink/yarn/YarnConfigurationITCase.java

```diff
@@ -187,6 +190,12 @@ public void testFlinkContainerMemory() throws Exception {
 					assertThat(
 						(double) taskManagerInfo.getHardwareDescription().getSizeOfJvmHeap() / (double) expectedHeadSize,
 						is(closeTo(1.0, 0.15)));
+
+					final int defaultTaskManagerMemoryMB = ConfigurationUtils.getTaskManagerHeapMemory(configuration).getMebiBytes();
+					YarnResourceManager.updateTaskManagerConfigAndCreateWorkerSlotProfiles(configuration, defaultTaskManagerMemoryMB, slotsPerTaskManager);
+					final int expectedManagedMemoryMB = MemorySize.parse(configuration.getString(TaskManagerOptions.MANAGED_MEMORY_SIZE)).getMebiBytes();
```

Comment:
I would suggest to calculate `expectedManagedMemoryMB` in a separate function, ideally w/o implicit modification of original `configuration`. Now it does not matter, just not to spoil something potentially added later.

Reply:
You are right, we should not modify the original `configuration` here.